src/Centreon/Domain/Authentication/UseCase/Authenticate.php

```diff
@@ -201,6 +203,30 @@ private function findProviderOrFail(string $providerConfigurationName): Provider
         return $authenticationProvider;
     }
 
+    /**
+     * Check if user is allowed to authenticate or throw an Exception.
+     *
+     * @param string $userName
+     * @throws AuthenticationException
+     */
+    private function authorizeUserToAuthenticateOrFail(string $userName): void
```

Comment:
Your method will try to authenticate the user or fail. Except that if the user is null, there is no exception.